http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProvider.scala

```diff
@@ -257,6 +257,35 @@ class SamAuthProvider[F[_]: OpenTelemetryMetrics](
           )
     } yield samUserInfo.userEmail
 
+  /**
+   * Confirm the Sam user is enabled (accepted terms of service etc).
+   * Raises an AuthProviderException if user does not exist or is disabled.
+   */
+  override def isUserEnabled(petOrUserInfo: UserInfo)(implicit ev: Ask[F, TraceId]): F[Boolean] =
```

Comment:
As implemented this never returns `false`.
I think this should either return `F[Unit]` and raise an error (and also name it `checkUserEnabled` or something); or return `F[Boolean]` and not raise an error for the `false` case.

Reply:
That makes sense, and speaks to a concern I had but couldn't articulate properly.